main/src/ca/uwaterloo/flix/api/lsp/provider/completion/Completion.scala

```diff
@@ -118,4 +127,32 @@ object Completion {
     */
   case class WithCompletion(name: String, priority: String, textEdit: TextEdit, documentation: Option[String],
                             insertTextFormat: InsertTextFormat) extends Completion
+
+
+  /**
+    * Represents an importNew completion (java constructors)
+    *
+    * @param name       the name of the completion.
+    * @param priority   the priority of the completion.
+    * @param textEdit   the edit which is applied to a document when selecting this completion.
+    */
+  case class ImportNewCompletion(name: String, priority: String, textEdit: TextEdit) extends Completion
+
+  /**
+    * Represents an importMethod completion (java methods)
+    *
+    * @param name     the name of the completion.
+    * @param priority the priority of the completion.
+    * @param textEdit the edit which is applied to a document when selecting this completion.
+    */
+  case class ImportMethodCompletion(name: String, priority: String, textEdit: TextEdit) extends Completion
```

Comment:
Why not store the actual field and method?

Reply:
java.lang.reflect.Field
java.lang.reflect.Method

and so forth

Reply:
And then just handle the completions in `Completion`, when it needs to be converted to a `CompletionItem`?

Reply:
The point is to avoid strings and use richer types. If you are completing a field it would be better to have that field.